torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2691,16 +2688,18 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::collective(
   work->outputs_ =
       std::make_shared<std::vector<at::Tensor>>(std::move(outputs));
 
-  if (avoidRecordStreams) {
+  if (avoidRecordStreams && inputs.size() == 1) {
     work->stashed_for_allocator_safety_ =
         std::make_shared<std::vector<at::Tensor>>();
-    work->stashed_for_allocator_safety_->push_back(input);
+    work->stashed_for_allocator_safety_->push_back(inputs[0]);
   }
```

Comment:
Stashing all `inputs`?

Reply:
Probably no, because in all-to-all we did that in preprocess.

Reply:
I think it is better to think from THIS function's perspective (to make it standalone).